planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp

```diff
@@ -1832,371 +1833,55 @@ void AvoidanceModule::addReturnShiftLineFromEgo(AvoidLineArray & sl_candidates)
 }
 
 bool AvoidanceModule::isSafePath(
-  const PathShifter & path_shifter, ShiftedPath & shifted_path, DebugData & debug) const
+  ShiftedPath & shifted_path, [[maybe_unused]] DebugData & debug) const
 {
-  const auto & p = parameters_;
+  const auto & p = planner_data_->parameters;
 
-  if (!p->enable_safety_check) {
+  if (!parameters_->enable_safety_check) {
     return true;  // if safety check is disabled, it always return safe.
   }
 
-  const auto & forward_check_distance = p->object_check_forward_distance;
-  const auto & backward_check_distance = p->safety_check_backward_distance;
-  const auto check_lanes =
-    getAdjacentLane(path_shifter, forward_check_distance, backward_check_distance);
-
-  auto path_with_current_velocity = shifted_path.path;
-
-  const size_t ego_idx = planner_data_->findEgoIndex(path_with_current_velocity.points);
-  utils::clipPathLength(path_with_current_velocity, ego_idx, forward_check_distance, 0.0);
-
-  constexpr double MIN_EGO_VEL_IN_PREDICTION = 1.38;  // 5km/h
-  for (auto & p : path_with_current_velocity.points) {
-    p.point.longitudinal_velocity_mps = std::max(getEgoSpeed(), MIN_EGO_VEL_IN_PREDICTION);
-  }
-
-  {
-    debug_data_.path_with_planned_velocity = path_with_current_velocity;
-  }
-
-  return isSafePath(path_with_current_velocity, check_lanes, debug);
-}
-
-bool AvoidanceModule::isSafePath(
-  const PathWithLaneId & path, const lanelet::ConstLanelets & check_lanes, DebugData & debug) const
-{
-  if (path.points.empty()) {
-    return true;
-  }
-
-  const auto path_with_time = [&path]() {
-    std::vector<std::pair<PathPointWithLaneId, double>> ret{};
-
-    float travel_time = 0.0;
-    ret.emplace_back(path.points.front(), travel_time);
+  const auto ego_predicted_path =
+    utils::avoidance::convertToPredictedPath(shifted_path.path, planner_data_, parameters_);
 
-    for (size_t i = 1; i < path.points.size(); ++i) {
-      const auto & p1 = path.points.at(i - 1);
-      const auto & p2 = path.points.at(i);
+  const auto ego_idx = planner_data_->findEgoIndex(shifted_path.path.points);
+  const auto is_right_shift = [&]() -> std::optional<bool> {
```

Comment:
@satoshi-ota
Like always some small question happenst to me.
This `is_right_shift` is the flag to judge ego-vehicle is shifting and judge if safety check should be excuted based on the information i guess.
If so, only shifting to right direction is only left side running country like Japan?
So `is_shifting` or `is_avoiding` is more suitable I thought.

Reply:
@kyoichi-sugahara Thanks for your question.

> This is_right_shift is the flag to judge ego-vehicle is shifting and judge if safety check should be excuted based on the information i guess.

Yes. If ego doesn't need to shift, the value will be `std::nullopt`.

> If so, only shifting to right direction is only left side running country like Japan?

No. I think there is a possiblity to meet the situation we have to shift right side on right-side traffic country. Additionaly, we sometimes met left shift situation in Odaiba.

Therefore, I think the variable `is_right_shift` needs to show following three case:

1. `is_right_shift: true` -> RIGHT side shift
2. `is_right_shift: false` -> LEFT side shift
3. `is_right_shift: std::nullopt`: NO shift lines exist

In conclusion, I suppose that it is better to use `is_right_shift`. What do you think?